Docs: Update GCP integration docs (#17097)
PrefectHQ/prefect: docs/integrations/prefect-gcp/index.mdx

---

`prefect-gcp` helps you leverage the capabilities of Google Cloud Platform (GCP) in your workflows.

> For example, you can run flows on Vertex AI or Cloud Run, read and write data to BigQuery and Cloud Storage, and retrieve secrets with Secret Manager. asdf 

Reviewer: ```suggestion
For example, you can run flows on Vertex AI or Cloud Run, read and write data to BigQuery and Cloud Storage, and retrieve secrets with Secret Manager.
```

### Install `prefect-gcp`

Install `prefect-gcp` as an extra of `prefect`. 
If you don't already have `prefect` installed, it will install the newest version of `prefect` as well.

<CodeGroup>

```bash pip
pip install -U "prefect[gcp]"
```

```bash uv
uv pip install -U "prefect[gcp]"
```

</CodeGroup>

If using BigQuery, Cloud Storage, Secret Manager, or Vertex AI, see [additional installation options](#install-extras).

#### Install extras
To install `prefect-gcp` with all additional capabilities, run the install command above and then run the following command:

<CodeGroup>

```bash pip
pip install -U "prefect-gcp[all_extras]"
```

```bash uv
uv pip install -U "prefect-gcp[all_extras]"
```

</CodeGroup>

> Or, install extras individually:

Reviewer: ```suggestion
Alternatively, you can install extras individually:
```

Author: Should we stick with the imperative style here?

Reviewer: Good point, imperative is probably better.

### Install `prefect-gcp`

> Install `prefect-gcp` as a dependency of `prefect`. 

Reviewer 2: ```suggestion
Install `prefect-gcp` as an extra of `prefect`. 
```

Author: Thanks!

pip install -U "prefect[gcp]"
```

If using BigQuery, Cloud Storage, Secret Manager, or Vertex AI, see [additional installation options](#install-extras).

#### Install extras
To install `prefect-gcp` with all additional capabilities, run the install command above and then run the following command:

```bash
pip install "prefect-gcp[all_extras]"
```

Or, install extras individually:

```bash
# Use Cloud Storage
pip install "prefect-gcp[cloud_storage]"

# Use BigQuery
pip install "prefect-gcp[bigquery]"

# Use Secret Manager
pip install "prefect-gcp[secret_manager]"

# Use Vertex AI
pip install "prefect-gcp[aiplatform]"
```

> 

Reviewer 3: could we do a `<CodeGroup>` here for `pip` and `uv` here as suggested here?

https://github.com/PrefectHQ/prefect/pull/17088#discussion_r1951265824

</Warning>

### BigQuery

Read data from and write to Google BigQuery within your Prefect flows.

Be sure to [install](#install-extras) `prefect-gcp` with the BigQuery extra.

```python
from prefect_gcp.bigquery import GcpCredentials, BigQueryWarehouse

gcp_credentials = GcpCredentials.load("BLOCK-NAME-PLACEHOLDER")

bigquery_block = BigQueryWarehouse(
    gcp_credentials = gcp_credentials,
    fetch_size = 1  # Optional: specify a default number of rows to fetch when calling fetch_many
)
bigquery_block.save("my-bigquery-block")
```

### Secret Manager
Manage secrets in Google Cloud Platform's Secret Manager.

```python
from prefect_gcp import GcpCredentials, GcpSecret
gcp_credentials = GcpCredentials.load("BLOCK-NAME-PLACEHOLDER")

gcp_secret = GcpSecret(
    secret_name = "your-secret-name",
    secret_version = "latest",
    gcp_credentials = gcp_credentials
)

gcp_secret.save("YOUR-GCP-SECRET-BLOCK-NAME")
```

### Cloud Storage
Create a block to interact with a GCS bucket.
```python
from prefect_gcp import GcpCredentials, GcsBucket

gcs_bucket = GcsBucket(
    bucket="your-bucket-name",

>     gcp_credentials=GcpCredentials.load("my-gcp-credentials")

Reviewer 3: can we be consistent with the case/style of the placeholders?

Reviewer: I like Google formatting approach to placeholders: https://developers.google.com/style/placeholders

If `service_account_file` is used, the provided path *must be available* in the container executing the flow.
</Warning>

### BigQuery

Read data from and write to Google BigQuery within your Prefect flows.

Be sure to [install](#install-extras) `prefect-gcp` with the BigQuery extra.

```python
from prefect_gcp.bigquery import GcpCredentials, BigQueryWarehouse

> gcp_credentials = GcpCredentials.load("CREDENTIALS-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
gcp_credentials = GcpCredentials.load("CREDENTIALS_BLOCK_NAME")
```

bigquery_block = BigQueryWarehouse(
    gcp_credentials = gcp_credentials,
    fetch_size = 1  # Optional: specify a default number of rows to fetch when calling fetch_many
)

> bigquery_block.save("BIGQUERY-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
bigquery_block.save("BIGQUERY_BLOCK_NAME")
```

Reviewer 3: @daniel-prefect block names may only contain hyphens, not underscores

Reviewer 4: Fair enough

```

### Secret Manager
Manage secrets in Google Cloud Platform's Secret Manager.

```python
from prefect_gcp import GcpCredentials, GcpSecret

> gcp_credentials = GcpCredentials.load("CREDENTIALS-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
gcp_credentials = GcpCredentials.load("CREDENTIALS_BLOCK_NAME")
```

gcp_secret = GcpSecret(

>     secret_name = "your-secret-name",

Reviewer: ```suggestion
    secret_name = "YOUR_SECRET_NAME",
```

    secret_version = "latest",
    gcp_credentials = gcp_credentials
)

> gcp_secret.save("SECRET-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
gcp_secret.save("SECRET_BLOCK_NAME")
```

```

### Cloud Storage
Create a block to interact with a GCS bucket.
```python
from prefect_gcp import GcpCredentials, GcsBucket

gcs_bucket = GcsBucket(
    bucket="BUCKET-NAME-PLACEHOLDER",

>     gcp_credentials=GcpCredentials.load("BIGQUERY-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
    gcp_credentials=GcpCredentials.load("BIGQUERY_BLOCK_NAME")
```

)

> gcs_bucket.save("GCS-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
gcs_bucket.save("GCS_BLOCK_NAME")
```

```

## Run flows on Google Cloud Run or Vertex AI

Run flows on [Google Cloud Run](https://cloud.google.com/run) or [Vertex AI](https://cloud.google.com/vertex-ai) to dynamically scale your infrastructure.

> Prefect Cloud offers [Google Cloud Run push work pools](/v3/deploy/infrastructure-examples/serverless). Push work pools submit runs directly to Google Cloud Run, instead of requiring a worker to actively poll for flow runs to execute.

Reviewer: Ventilated prose

```suggestion
Prefect Cloud offers [Google Cloud Run push work pools](/v3/deploy/infrastructure-examples/serverless).
Push work pools submit runs directly to Google Cloud Run, instead of requiring a worker to actively poll for flow runs to execute.
```

@flow
def bigquery_flow():
    all_rows = []

>     gcp_credentials = GcpCredentials.load("CREDENTIALS-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
    gcp_credentials = GcpCredentials.load("CREDENTIALS_BLOCK_NAME")
```

    file_path = Path("test-example.txt")
    file_path.write_text("Hello, Prefect!")

>     gcp_credentials = GcpCredentials.load("CREDENTIALS-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
    gcp_credentials = GcpCredentials.load("CREDENTIALS_BLOCK_NAME")
```

    gcs_bucket = GcsBucket(

>         bucket="BUCKET-NAME-PLACEHOLDER",

Reviewer: ```suggestion
        bucket="BUCKET_NAME",
```

@flow
def secret_manager_flow():

>     gcp_credentials = GcpCredentials.load("CREDENTIALS-BLOCK-NAME-PLACEHOLDER")

Reviewer: ```suggestion
    gcp_credentials = GcpCredentials.load("CREDENTIALS_BLOCK_NAME")
```

    secret_manager_flow()
```

## Resources

For assistance using GCP, consult the [Google Cloud documentation](https://cloud.google.com/docs).

GCP can also authenticate without storing credentials in a block.

> See the [Third-party Secrets docs](/v3/develop/secrets) for an analogous example that uses AWS Secrets Manager and Snowflake.

Reviewer: ```suggestion
See [Access third-party secrets](/v3/develop/secrets) for an example that uses AWS Secrets Manager and Snowflake.
```

> Refer to the `prefect-gcp` SDK documentation linked in the sidebar to explore all the capabilities of the `prefect-gcp` library.

Reviewer: I suggest linking directly to the intended page instead of describing where to find it.

```suggestion
Refer to [prefect-gcp](/integrations/prefect-gcp/) to explore all of the capabilities of the `prefect-gcp` library.
```
